### PR TITLE
Implemented IID Delete API

### PR DESCRIPTION
**Pull request by hiranya911**
A new API that enables deleting instance IDs associated with a Firebase project.

```
client, err := app.InstanceID(ctx)
if err != nil {
  // handle error
}

if err := client.DeleteInstanceID(iid); err != nil {
  // handle error
}
```

go/firebase-admin-iid

### CODE REVIEW

File: iid/iid.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"google.golang.org/api/transport"
+
+	"firebase.google.com/go/internal"
+
+	"golang.org/x/net/context"
+)
+
+const iidEndpoint = "https://console.firebase.google.com/v1"
+
+// Client is the interface for the Firebase instance ID service.
```
Comment (bklimt) [inline iid/iid.go]: I would capitalize "instance" here.

File: iid/iid.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+type Client struct {
+	endpoint string
```
Comment (bklimt) [inline iid/iid.go]: Maybe add a comment that this is here only for replacing it during testing?

File: iid/iid_test.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	client.endpoint = ts.URL
+	if err := client.DeleteInstanceID(ctx, "test-iid"); err != nil {
+		t.Errorf("DeleteInstanceID() = %v; want nil", err)
+	}
+
+	if tr == nil {
+		t.Fatalf("Request = nil; want non-nil")
+	}
+	if tr.Method != "DELETE" {
+		t.Errorf("Method = %q; want = %q", tr.Method, "DELETE")
```
Comment (bklimt) [inline iid/iid_test.go]: On these, it should probably be "want %q" instead of "want = %q".

File: integration/iid/iid_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	os.Exit(m.Run())
+}
+
+func TestNonExisting(t *testing.T) {
+	err := client.DeleteInstanceID(context.Background(), "non-existing")
+	if err == nil {
+		t.Errorf("DeleteInstanceID(non-existing) = nil; want error")
+	}
+	if !strings.HasPrefix(err.Error(), "http error status: 404") {
+		t.Errorf("DeleteInstanceID(non-existing) = %v; want = 404", err)
```
Comment (bklimt) [inline integration/iid/iid_test.go]: Same here.

File: iid/iid.go (hunk 1 above)
Comment (hiranya911) [reply iid/iid.go]: Done

File: iid/iid.go (hunk 2 above)
Comment (hiranya911) [reply iid/iid.go]: Done

File: iid/iid_test.go (hunk 3 above)
Comment (hiranya911) [reply iid/iid_test.go]: We've been following this convention in this repo. I would stick to it for now. We can later discuss if we should revise it.

**Pull request merged**